src/providers/cloudstack/network.rs

```diff
@@ -30,15 +30,19 @@ impl CloudstackNetwork {
     }
 
     fn endpoint_for(&self, key: &str) -> String {
-        format!("http://{}/latest/meta-data/{}", self.server_address, key)
+        format!("{}/latest/meta-data/{}", self.server_address, key)
     }
 
-    fn get_dhcp_server_address() -> Result<IpAddr> {
+    fn get_dhcp_server_address() -> Result<String> {
+        if cfg!(test) {
```

Comment:
We usually make two separate function implementations, one `test` and one `not(test)`.

Reply:
Yep, but than `cargo test` complains about unused imports and never-called functions. I wasn't sure if it's Ok to see such warnings during `test`.